python/metricq/connection_watchdog.py

```diff
@@ -104,6 +117,26 @@ def set_closed(self):
         """Stop the connection watchdog task if it is running.
         """
         if self._watchdog_task:
+            if self._watchdog_task.done():
+                try:
+                    logger.warn(
+                        "Watchdog task {} already done with result: {}",
+                        self.connection_name,
+                        self._watchdog_task.result(),
+                    )
+                except Exception as e:
+                    logger.error(
+                        "Watchdog task {} already done with exception: {}",
+                        self.connection_name,
+                        e,
+                    )
+
             self._watchdog_task.cancel()
-            await self._watchdog_task
+
```

Comment:
If I understand this correctly, the watchdog task is only considered to have stopped successfully if it raises a `CancelledError`?

Reply:
I just follow the normal (symmetric) path of cancellation here instead of converting the cancellation into a regular completion within the task.